packages/editor/src/services/ModeServices.ts

```diff
@@ -0,0 +1,102 @@
+import { createState, useState } from '@speigg/hookstate'
+
+import { store } from '@xrengine/client-core/src/store'
+
+type ModeServiceStateType = {
+  renderModeChanged: boolean
+  playModeChanged: boolean
+  flyModeChanged: boolean
+  transformMode: string
+  transformModeChanged: boolean
+  transformSpaceModeChanged: boolean
+  transformPivotModeChanged: boolean
+  snapSettingsChanged: boolean
+}
+
+const state = createState<ModeServiceStateType>({
+  renderModeChanged: false,
+  playModeChanged: false,
+  flyModeChanged: false,
+  transformMode: '',
+  transformModeChanged: false,
```

Comment:
Since there is `transformMode` state, having `transformModeChanged` is redudent in my opinion. We can use value of `transformMode` to determine the change which was not possible before this refactor hence we had 2 properties.

Reply:
Will update this flow, Thanks.